neonvm/controllers/virtualmachine_controller.go

```diff
@@ -955,6 +990,44 @@ func (r *VirtualMachineReconciler) podForVirtualMachine(
 	return pod, nil
 }
 
+func (r *VirtualMachineReconciler) sshSecretForVirtualMachine(virtualmachine *vmv1.VirtualMachine) (*corev1.Secret, error) {
+	secret, err := sshSecretSpec(virtualmachine)
+	if err != nil {
+		return nil, err
+	}
+
+	// Set the ownerRef for the Secret
+	// More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/owners-dependents/
+	if err := ctrl.SetControllerReference(virtualmachine, secret, r.Scheme); err != nil {
+		return nil, err
+	}
+	return secret, nil
+}
+
+func sshSecretSpec(virtualmachine *vmv1.VirtualMachine) (*corev1.Secret, error) {
+	// with bitSize = 2048 it takes ~100ms
```

Comment:
I know this is still WIP - wanted to comment on this.

100ms is actually quite a lot of time for the global controller to spend CPU-bound during VM startup / creation. Sometimes (particularly, at midnight UTC) the console / control plane creates many VMs all at once.
(See e.g. [this logs query](https://neonprod.grafana.net/explore?schemaVersion=1&panes=%7B%22EL3%22:%7B%22datasource%22:%22grafanacloud-logs%22,%22queries%22:%5B%7B%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22grafanacloud-logs%22%7D,%22editorMode%22:%22code%22,%22expr%22:%22sum%20by%28cluster%29%20%28count_over_time%28%5Cn%7Bjob%3D%5C%22integrations%2Fkubernetes%2Feventhandler%5C%22%7D%5Cn%20%20%20%20%7C%20logfmt%5Cn%20%20%20%20%7C%20sourcecomponent%20%3D%20%5C%22virtualmachine-controller%5C%22%5Cn%20%20%20%20%7C%20msg%20%3D~%20%5C%22VirtualMachine%20.%2A%20created,%20pod%20.%2A%5C%22%5Cn%20%20%20%20%5B1s%5D%5Cn%29%29%22,%22queryType%22:%22range%22,%22refId%22:%22A%22,%22step%22:%221s%22,%22legendFormat%22:%22%7B%7Bcluster%7D%7D%22%7D%5D,%22range%22:%7B%22from%22:%221704326340000%22,%22to%22:%221704326460000%22%7D%7D%7D&orgId=1) that visualizes the number of VMs created each second [kind of - it's close enough])

From what I can remember (+ confirmed with a quick search, and then benchmarking locally), elliptic curve keys appear to be substantially quicker to generate for the same strength. For example, on my machine, 4096-bit RSA keys ~1s to generate (2048-bit is ~85ms, 3072-bit is ~390ms), whereas Ed25519 keys take ~13 µs (~16 µs if you include the marshaling).